performance/src/performance/timer.cpp

```diff
@@ -0,0 +1,41 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Library include(s).
+#include "traccc/performance/timer.hpp"
+
+// System include(s).
+#include <algorithm>
+
+namespace traccc::performance {
+
+/// Start time measurement
+/// @param timer_name name to be printed out identifying what is measured
+/// @param t_info shared_ptr to timing_info where to store timings
+timer::timer(const std::string_view timer_name, timing_info* t_info)
+    : m_name(timer_name), m_timing_info(t_info) {
+    m_start = std::chrono::high_resolution_clock::now();
```

Comment:
You should just do this with the same formalism that you use to initialise the other two members. I.e.

```c++
timer::timer(const std::string_view timer_name, timing_info* t_info)
    : m_start(std::chrono::high_resolution_clock::now()),
      m_name(timer_name), m_timing_info(t_info) {}
```